lib/capabilities.go

```diff
@@ -0,0 +1,121 @@
+package ircbnc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/goshuirc/irc-go/ircmsg"
+)
+
+type CapManager struct {
+	Supported            map[string]string
+	FnsInitListener      map[string]func(*Listener)
+	FnsMessageToClient   []func(*Listener, *ircmsg.IrcMessage) bool
+	FnsMessageFromClient []func(*Listener, *ircmsg.IrcMessage) bool
+}
+
+var Capabilities CapManager
+
+func init() {
+	Capabilities = CapManager{
+		Supported:       make(map[string]string),
+		FnsInitListener: make(map[string]func(*Listener)),
+	}
+
+	CapAwayNotify(&Capabilities)
+	CapServerTime(&Capabilities)
+}
+
+// AsString returns a list ready to send to the client of all our CAPs
+func (caps *CapManager) AsString() string {
```

Comment:
This should just be named `String` imo (or otherwise maybe more descriptive as `SupportedCapsString`), matches what Go tends to do for naming of these sorta functions.